### ISSUE TITLE
[bug] No options for build requirements

### ISSUE THREAD

**Issue by Sil3ntStorm** [conan-io/conan#11508]
### Environment Details (include every applicable attribute)
  * Operating System+version: Windows 10 x64, Arch Linux
  * Compiler+version: MSVC 2022, clang14
  * Conan version: 1.49.0
  * Python version: 3.9.7

### Steps to reproduce (Include if Applicable)
Have a conan recipe that has dependencies on other conan packages, which are only there if the package is built, rather than consumer requirements and try to set or check options for these dependencies.

```
from conans import ConanFile, tools
from conans.errors import ConanInvalidConfiguration

class BugDemoConan(ConanFile):
    name = 'bug_demo_project'
    version = '0.2.6'
    author = 'Nobody <nobody@cares.invalid>'
    url = 'www.bugz.com'
    description = 'Bug Demo'
    license = 'Unlicense'
    settings = 'os', 'compiler', 'build_type', 'arch'
    generators = 'cmake'
    build_requires = (
        'nlohmann_json/3.10.5',
        'ms-gsl/4.0.0',
        'spdlog/1.10.0',
    )

    def configure(self):
        self.options['fmt'].header_only = True
        self.options['spdlog'].header_only = True
        if self.settings.os == 'Windows':
            self.options['spdlog'].wchar_support = True
            self.options['spdlog'].wchar_filenames = True

    def validate(self):
        if self.options['fmt'].header_only != True:
            raise ConanInvalidConfiguration('fmt should be used in header_only mode')
        if self.options['spdlog'].header_only != True:
            raise ConanInvalidConfiguration('spdlog should be used in header_only mode')
        if self.settings.os == 'Windows' and not (self.options['spdlog'].wchar_filenames and self.options['spdlog'].wchar_support):
            raise ConanInvalidConfiguration('spdlog should be used with wchar_support and wchar_filenames on Windows')

```

### Logs (Executed commands with output) (Include/Attach if Applicable)
```
conan create .
Exporting package recipe
bug_demo_project/0.2.6: A new conanfile.py version was exported
bug_demo_project/0.2.6: Folder: <snip>\.conan\data\bug_demo_project\0.2.6\_\_\export
bug_demo_project/0.2.6: Exported revision: b215e3b34385bf036a8001d65ff71733
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.runtime=MD
compiler.version=17
os=Windows
os_build=Windows
[options]
[build_requires]
[env]

ERROR: bug_demo_project/0.2.6: Error in validate() method, line 27
        if self.options['fmt'].header_only != True:
        ConanException: option 'header_only' doesn't exist
Possible options are []
```

**Comment by memsharded** [comment]
Hi @Sil3ntStorm 

We are aware that there is a limitation regarding options for build-requires.

In Conan 1.x the best approach could be to leave the options definitions in the "build" profile.

Conan 2.0 implements a ``self.tool_requires("tool/version", options={"tool_option": "value"})`` to be able to specify options for an individual requires.

**Comment by Sil3ntStorm** [comment]
While that might work, it still does not allow to validate these settings are actually present and fail the build otherwise. But I guess there is no way around that.

Thanks.

Ideally in the future you would be able to validate that these settings are set correctly somewhere.

**Comment by memsharded** [comment]
Sorry, I am a bit confused, the issue above is about options, while now you are talking about ``settings``, which is a completely different thing. I would need to understand better what you mean about that expected validation, not clear.

**Comment by Sil3ntStorm** [comment]
I'm sorry I meant options, since somehow it seems all the settings are gone, the above validate function does not work. Which is all I was trying to say.
So I will have to trust that the profile is setup correctly.

**Comment by memsharded** [comment]
Validation can be done with real values of options dependencies, but it needs to be accessed with ``self.dependencies["mydep"].options``. That has helpers to address exactly the context that you want like ``self.dependencies.build["mydep"].options`` is different than ``self.dependencies.host["mydep"].options``

Have a look to https://docs.conan.io/en/latest/reference/conanfile/dependencies.html

**Comment by memsharded** [comment]
Conan 2 now contains several mechanisms to define options in build-requires:

- ``self.tool_requires(..., options={})``
- ``default_build_options``

Closing as solved in 2.0
